pkg/provisioner/ironic/raid.go

```diff
@@ -138,29 +140,56 @@ func buildTargetSoftwareRAIDCfg(volumes []metal3v1alpha1.SoftwareRAIDVolume) (lo
 	return
 }
 
+// hwRaidConfigured returns true if HardwareRAIDVolumes is non-nil.
+// Note that it still returns true if it is an empty list - in this case we
+// still want to clear the RAID settings. When it is nil, we just leave the
+// existing settings in place.
+func hwRaidConfigured(config *metal3v1alpha1.RAIDConfig) bool {
```

Comment:
Maybe we can check the value of `ironicNode.RAIDConfig` to determine whether RAID has been configured.

Reply:
The ironic database could disappear at any time, so the k8s CR has to be the source of truth.

Reply:
What do you think of adding a new member to `prepareData` to store the current RAID configuration, and blanking the the new member when clean failed to avoid deleting the saved RAID configuration? See here for [details](https://github.com/metal3-io/baremetal-operator/pull/908/files#diff-a69d9029388ab766ed36b32180145f52785a9d4a153775510dbddfa928a72e1cR759-R771).